app/src/sharedTest/java/org/oppia/android/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -222,20 +223,25 @@ class ExplorationActivityTest {
   @Inject
   lateinit var fakeAccessibilityService: FakeAccessibilityService
 
+  @Inject
+  lateinit var profileManagementController: ProfileManagementController
+
   private val internalProfileId: Int = 0
 
+  private val afternoonUtcTimestampMillis = 1556101812000
+
   @Before
   fun setUp() {
     Intents.init()
     setUpTestApplicationComponent()
     testCoroutineDispatchers.registerIdlingResource()
     profileTestHelper.initializeProfiles()
     fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    testCoroutineDispatchers.unregisterIdlingResource()
```

Comment:
Why was this changed? Generally the idling resources are needed until the test finishes.

Reply:
Unfortunately, this seems to not work as intended when running the tests on Espresso. This also applies to other large tests like StateFragment Test.

The IdlingResource always times out. From my println logs, I think there is a loop somewhere.

```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: androidx.test.espresso.IdlingResourceTimeoutException: Wait for [TestCoroutineDispatcherIdlingResource] to become idle timed out
at androidx.test.espresso.Espresso.onIdle(Espresso.java:342)
at androidx.test.espresso.Espresso.onIdle(Espresso.java:361)
at org.oppia.android.testing.threading.TestCoroutineDispatchersEspressoImpl.advanceUntilIdle(TestCoroutineDispatchersEspressoImpl.kt:45)
at org.oppia.android.testing.threading.TestCoroutineDispatchersEspressoImpl.runCurrent(TestCoroutineDispatchersEspressoImpl.kt:34)
at org.oppia.android.app.player.exploration.ExplorationActivityTest.testExplorationActivity_loadingAudio_progressbarIsDisplayed(ExplorationActivityTest.kt:1321)
...
androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:527)
at org.oppia.android.testing.OppiaTestRule$apply$1.evaluate(OppiaTestRule.kt:47)
at org.oppia.android.testing.junit.InitializeDefaultLocaleRule$apply$1.evaluate(InitializeDefaultLocaleRule.kt:53)
...
Caused by: java.util.concurrent.ExecutionException: androidx.test.espresso.IdlingResourceTimeoutException: Wait for [TestCoroutineDispatcherIdlingResource] to become idle timed out
at java.util.concurrent.FutureTask.report(FutureTask.java:123)
at java.util.concurrent.FutureTask.get(FutureTask.java:193)
at androidx.test.espresso.Espresso.onIdle(Espresso.java:334)
```

Reply:
I will file a seperate issue to investigate this further

Reply:
Reverting these changes and filing https://github.com/oppia/oppia-android/issues/5379 for investigation instead.